File: cpp/problem38.cpp
#include <mystd.h>
using namespace std;
using namespace MYSTD;

class Solution {
public:
    string countAndSay(int n) {
		string s("1");
		for (int i = 0; i < n-1; ++i){
			countAndSayOnce( s );
		} 
		return s;
    }
    void countAndSayOnce(string & s){
    	string temp = "";
    	int i = 0, n = s.size();
    	for (int j = 1; j < n; ++j){
    		if( s[j] != s[i] ){
    			temp += to_string( j - i ) + s[i];
    			i = j;
    		}
    	}
    	temp += to_string( n - i ) + s[i];
    	s = std::move( temp );
    }
};


int main(int argc, char const *argv[]){
	Solution sol;
	for (int i = 1; i <= 30; ++i){
		cout <<  sol.countAndSay(i) << endl;	
	}
	return 0;
}

File: java/problem38.java
class Solution {
    public String countAndSay(int n) {
        String result=new String("1");
        String temp;
        for(int i=1;i<n;i++){
            temp="";
            int current=0;int sweep;
            for(sweep=0;sweep<result.length();sweep++){
                if(result.charAt(sweep)!=result.charAt(current)){
                    temp+=String.valueOf(sweep-current);
                    temp+=result.charAt(current);
                    current=sweep;
                }
            }
            temp+=String.valueOf(sweep-current);
            //System.out.println(temp);
            temp+=result.charAt(current);
            //System.out.println(temp);
            result=temp;
        }
        return result;
    }
}

public class problem38{
     public static void main(String args[]){
         Solution test=new Solution();
         System.out.println(test.countAndSay(5));
    }
}